src/client/Microsoft.Identity.Client/AccountId.cs

```diff
@@ -63,14 +63,14 @@ internal static AccountId ParseFromString(string str)
             {
                 return null;
             }
-            string[] elements = str.Split('.');
+            int lastIndexOfDot = str.LastIndexOf('.');
 
-            if (elements.Length == 1)
+            if (lastIndexOfDot == -1)
             {
                 return new AccountId(str); //Account id is from ADFS; no . in the string
             }
 
-            return new AccountId(str, elements[0], elements[1]);
+            return new AccountId(str, str.Substring(0, lastIndexOfDot), str.Substring(lastIndexOfDot + 1));
```

Comment:
Very cool!

Can you add more tests around these string operations? Lots of "off by 1" errors possible. This method is static so should be very easy to test.

Reply:
Added tests for parsing